core/listener.go

```diff
@@ -32,8 +35,10 @@ var tracer = otel.Tracer("core/listener")
 type Listener struct {
 	fetcher *BlockFetcher
 
-	construct header.ConstructFn
-	store     *eds.Store
+	headerConstruct header.ConstructFn
+	edsConstruct    eds.ConstructFn
```

Comment:
Looks like those are configurable to be allow mocking for tests. I wonder if at this point It would be worth to consider making a single function Interface instead of exact types. 

Just thinking, if we can make it a bit more go-way later. lets not do it in this PR 🙂